### ISSUE TITLE
Critical Error on previously freed Level Scene when using PCamRoom Nodes in runtime instantiated and freed scenes.

### ISSUE THREAD

**Issue by HiIAmBarney** [daz-b-like/ProCam2D_Godot4.x#5]
**Describe the bug**
Freeing a Scene with a PCamRoom Node throws an Error: 
"Invalid access to property or key 'global_position' on a base object of type 'previously freed'."

**To Reproduce**
1. Create a Level Scene with a PCamRoom Node Setup. (Player and ProCam are not necessarily part of the instantiated Level Scene)
2. Instantiate the new Level Scene (With a PCamRoom Node Setup) and Change the current Scene to the new Scene
3. Free the previously instantiated Scene

**Expected behavior**
The newly instantiated Scene should have its PCamRoom Node overwrite the current PCamRoom Node or freeing the previously instantiated PCamRoom Node should safely disconnect on leaving the tree.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Godot Version 4.3
 - Plugin Version ProCam2D_Godot4.x v 1.0 (latest)

**Additional context**
This is a follow-up issue on the PCamRoom instantiation issue.

**Comment by daz-b-like** [comment]
Hey @HiIAmBarney . I added some safety checks. Here's the [new version](https://github.com/daz-b-like/ProCam2D_Godot4.x/archive/5256a0a89a83ddf6c316978c368b2665e7396ebd.zip). It should work fine now.. Thanks again for reporting. If you have any issues let me know. Good day.

**Comment by HiIAmBarney** [comment]
There IS in fact an Issue with this.
When I switch rooms now, the camera Zooms in (and stays Zoomed until I leave the PCamRooms influence), even though the Zoom factor is set to 1 within the Node. 
I added some Screenshots, though, pardon my placeholder Art, it's from a Game Jam I participated in, that got scrapped mid-project, I am working on realizing it.

This is the Room 1 (In-Game)
![Room1](https://github.com/user-attachments/assets/306ac0eb-bf7c-4aea-96db-2d596984ecb1)

This is the Room 2 (In-Game)
![Room2](https://github.com/user-attachments/assets/0dfeb52a-bbd7-424d-9aca-3bf801d0714b)

This is the Room 2 (In_Editor)
![Room2InEditor](https://github.com/user-attachments/assets/d880cbb8-4811-46d4-9a1e-b36050c7517b)


Edit: Don't mind the Debug Velocity and Positional Values, these are not hooked up atm.

**Comment by daz-b-like** [comment]
I'm not really clear with how you intend this to work so pardon me if I'm wrong. I've noticed the height of room 2 is 192px. I made it so that the rooms never show what's outside their boundaries. In this case the room automatically adjusts the PCam zoom level in order to fit its height to the screen to avoid showing what's outside its boundary and its zoom factor becomes relative to the new zoom level. To avoid auto zooming, the rooms width and height need to be equal or more than the viewports width and height. Try setting both rooms the same height maybe?
